geiger/src/lib.rs

```diff
@@ -35,58 +51,47 @@ impl fmt::Display for ScanFileError {
     }
 }
 
-/// Scan result for a single `.rs` file.
-#[derive(Clone, Debug, Default, PartialEq)]
-pub struct RsFileMetrics {
-    /// Metrics storage.
-    pub counters: CounterBlock,
-
-    /// This file is decorated with `#![forbid(unsafe_code)]`
-    pub forbids_unsafe: bool,
-}
-
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub enum IncludeTests {
-    Yes,
-    No,
-}
-
-struct GeigerSynVisitor {
-    /// Count unsafe usage inside tests
-    include_tests: IncludeTests,
-
-    /// The resulting data from a single file scan.
-    metrics: RsFileMetrics,
-
-    /// The number of nested unsafe scopes that the GeigerSynVisitor are
-    /// currently in. For example, if the visitor is inside an unsafe function
-    /// and inside an unnecessary unsafe block inside that function, then this
-    /// number should be 2. If the visitor is outside unsafe scopes, in a safe
-    /// scope, this number should be 0.
-    /// This is needed since unsafe scopes can be nested and we need to know
-    /// when we leave the outmost unsafe scope and get back into a safe scope.
-    unsafe_scopes: u32,
+fn file_forbids_unsafe(f: &syn::File) -> bool {
+    use syn::AttrStyle;
+    use syn::Meta;
+    use syn::MetaList;
+    use syn::NestedMeta;
+    f.attrs
+        .iter()
+        .filter(|a| matches!(a.style, AttrStyle::Inner(_)))
+        .filter_map(|a| a.parse_meta().ok())
+        .filter(|meta| match meta {
+            Meta::List(MetaList {
+                path,
+                paren_token: _paren,
+                nested,
+            }) => {
+                if !path.is_ident("forbid") {
+                    return false;
+                }
+                nested.iter().any(|n| match n {
+                    NestedMeta::Meta(Meta::Path(p)) => {
+                        p.is_ident("unsafe_code")
+                    }
+                    _ => false,
+                })
+            }
+            _ => false,
+        })
+        .count()
+        > 0
```

Comment:
Since CI passes this is following rustfmt standard, but still very strange formatting style to use a newline here imho... :)

Reply:
I actually went back and had a go at setting it explicitly to `.count() > 0` but it appears cargo fmt seems to update it to be like this.
Do you think its worth looking into the cargo fmt settings, or do you reckon it's OK like this?

Reply:
If this is what rustfmt does, let's go with that.